application/helpers/remotecontrol/remotecontrol_handle.php

```diff
@@ -573,16 +573,20 @@ public function export_statistics($sSessionKey, $iSurveyID, $docType = 'pdf', $s
         $aSummary = createCompleteSGQA($iSurveyID, $oAllQuestions, $sLanguage);
 
         $helper = new statistics_helper();
+        $tempDir = Yii::app()->getConfig("tempdir");
         switch ($docType) {
             case 'pdf':
-                $sTempFile = sanitize_filename(
-                    $helper->generate_statistics($iSurveyID, $aSummary, $aSummary, $graph, $docType, 'F', $sLanguage)
+                $sTempFile = $tempDir . DIRECTORY_SEPARATOR . sanitize_filename(
```

Comment:
why sanitize_filename since [helper->generate_statistics ](https://github.com/LimeSurvey/LimeSurvey/blob/master/application/helpers/admin/statistics_helper.php#L3796) already set sanitized filename. No reason ro sanitize a generated filename, no user entered.

If  needed : need to be done in helpe.

I think best is to remove sanitizer.

And then : no need extra var here
https://github.com/LimeSurvey/LimeSurvey/blob/de071e01d299b41cc79cc4067f06760ef351da31/application/helpers/admin/statistics_helper.php#L3884
https://github.com/LimeSurvey/LimeSurvey/blob/de071e01d299b41cc79cc4067f06760ef351da31/application/helpers/admin/statistics_helper.php#L4079

Reply:
Sonar_cloud make error. We really don'(t need to sanitize this var.

Reply:
Well, remove it and check it Sonarcloud thinks it is a security issue.

Reply:
We don't have way to add comments for sonar cloud ? Here :

1. it's clearly NOT a security issue
2. The fix apply for sonar_cloud create bug
3. The fix make the system more complex

Reply:
Seems soundcloud is ok with this?
https://sonarcloud.io/project/security_hotspots?id=LimeSurvey_LimeSurvey&pullRequest=3216

Reply:
:+1: